mrc-ide/packit
# mrc-4561 implement authentication

This PR serves as a proof of concept for implementing basic packet authentication. It lays the foundation for future enhancements, including token validation, fine-grained access control, and additional business logic as needed. I'm seeking a preliminary review before proceeding with test implementation.

TODO: 
1. Add browser test
2. Add logout endpoint in backend
3. Call logout endpoint in front end to destroy backend securityContext
4. Return custom message for unauthenticated handler in backend

To test:

1. Create a GitHub App
2. Use the below config
3. update api config.property
5. set GitHub secret and client id
6. run app

See below for a screenplay and GitHub config 

![Screenshot 2023-09-14 at 18 17 14](https://github.com/mrc-ide/packit/assets/4240748/fc063048-1b85-49af-a0b0-f492d3c2c2e2)




https://github.com/mrc-ide/packit/assets/4240748/fc73aa74-5200-41d5-9b2e-41bce27f85f9
